sdk/messaging/azservicebus/internal/utils/retrier.go

```diff
@@ -51,7 +51,14 @@ func Retry(ctx context.Context, eventName log.Event, operation string, fn func(c
 		if i > 0 {
 			sleep := calcDelay(ro, i)
 			log.Writef(eventName, "(%s) Retry attempt %d sleeping for %s", operation, i, sleep)
-			time.Sleep(sleep)
+
+			sleepCtx, cancelSleep := context.WithTimeout(ctx, sleep)
```

Comment:
This is the typical pattern we use elsewhere.

```go
select {
case <-ctx.Done():
	return ctx.Err()
case <-time.After(sleep):
	// sleep elapsed
}
```

Reply:
Yeah, I was aware of that one, but I did think this was pretty nice too.

I'll change it so we're consistent.

Reply:
Done! One nice advantage of the default approach is that we don't have to check ctx.Err() vs our local context. So I think it's clearer now.